buildenv/jenkins/common/variables-functions

```diff
@@ -0,0 +1,249 @@
+* if the map contains no mapping for the key. 
+*/
+def get_value(MAP, KEY) {
+    for (item in MAP) {
+        if ("${item.key}" == "${KEY}") {
+            return "${item.value}"
+        }
+    }
+
+    return ''
+}
+
+/*
+* Returns Jenkins credentials ID required to connect to GitHub using 
+* the ssh protocol.
+* Returns empty string if no values is provided in the variables file (not 
+* required for public repositories).
+*/
+def get_git_user_credentials_id () {
+    if (VARIABLES.user != null) {
+        return VARIABLES.user.credentials_id
+    }
+
+    return ''
+}
+
+/*
+* Sets the NODE variable to the labels identifying nodes by platform suitable 
+* to run a Jenkins job.
+*/
+def set_node(job_type) { 
+    // fetch labels (space separated string) for given platform/spec
+    labels = get_value(VARIABLES."${SPEC}".node_labels, job_type)
+    NODE = labels.replaceAll(" ", "&&")
+}
+
+/*
+* Set the RELEASE variable with the value provided in the variables file.
+*/
+def set_release() {
+    RELEASE = get_value(VARIABLES."${SPEC}".release, SDK_VERSION)
+}
+
+/*
+* Set the JDK_FOLDER variable with the value provided in the variables file.
+*/
+def set_jdk_folder() {
+    JDK_FOLDER = get_value(VARIABLES.jdk_image_dir, SDK_VERSION)
+}
+
+/*
+* Sets variables for a job that builds the OpenJ9 extensions for OpenJDK for a 
+* given platform and version.
+*/
+def set_build_variables() {
+    set_repos_variables()
+
+    // fetch values per spec and Java version from the variables file
+    BOOT_JDK = get_value(VARIABLES."${SPEC}".boot_jdk, SDK_VERSION)
+    FREEMARKER = VARIABLES."${SPEC}".freemarker
+    set_release()
+    set_jdk_folder()
+
+    try{
+        SDK_PREFIX = "OpenJ9-JDK${SDK_VERSION}-PR${ghprbPullId}-${SPEC}-"
+    } catch (e) {
+        SDK_PREFIX = "OpenJ9-JDK${SDK_VERSION}-${SPEC}-" 
+    }
+
+    SDK_SUFFIX = ".tar.gz"
+    TEST_PREFIX = "test-"
+    TEST_SUFFIX = ".tar.gz"
+}
+
+/*
+* Sets variables for a test job on a given platform.
+*/
+def set_test_variables() {
+    set_release()
+    set_jdk_folder()
+    JAVA_VERSION = "SE" + "${SDK_VERSION}" + "0"
+    TEST_DEPENDENCIES_JOB_NAME = VARIABLES.test_dependencies_job_name
+}
+
+/*
+* Initializes all of the required variables for a Jenkins job by given job type.
+*/
+def set_job_variables(job_type) {
+    // initialize VARIABLES
+    parse_variables_file()
+
+    // fetch credentials required to connect to GitHub using ssh protocol
+    USER_CREDENTIALS_ID = get_git_user_credentials_id()
+
+    switch (job_type) {
+        case "build":
+            // set the node the Jenkins build would run on
+            set_node('build')
+            // set variables for a build job
+            set_build_variables()
+            break
+        case "test":
+            // set the node the tests would run on
+            set_node('test')
+            // set variables for a test job
+            set_test_variables()
+            break;
```

Comment:
this break ends with a `;` which is inconsistent with the other cases

Reply:
removed